File: Windows/OpenIoT.App/OpenIoT.Lib/Board/Transmission/ITransmissionChannel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenIoT.Lib.Board.Transmission
{
    public interface ITransmissionChannel
    {
        string Name { get; }

        void Write(byte[] buffer, int offset, int length);
        int Read(byte[] buffer, int offset, int length);

        bool Open();
        bool Close();

        bool IsOpen();
    }
}

File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/board/transmission/ITransmissionChannel.java
package com.palitri.openiot.construction.framework.board.transmission;

public interface ITransmissionChannel
{
    void write(byte[] buffer, int offset, int length);
    int read(byte[] buffer, int offset, int bufferLength);

    boolean open();
    boolean close();

    boolean isOpened();
}
